### PR TITLE
RangeCalendar. fix starttime and endtime is same day.

### PR DESCRIPTION
**Pull request by kumaw**
fix starttime and endtime is same day. if hours are different ,userSettingDisabledTime is not work.

https://github.com/ant-design/ant-design/issues/8915

### CODE REVIEW

**Comment by yesmeck** [comment]
@kumaw Could you add a test case?

File: src/RangeCalendar.js [hunk 1]
```diff
@@ -322,6 +322,8 @@ const RangeCalendar = createReactClass({
       const minutes = startValue.minute();
       const second = startValue.second();
       let { disabledHours, disabledMinutes, disabledSeconds } = userSettingDisabledTime;
+      const oldDisabledMinutes = disabledMinutes ? disabledMinutes() : [];
```
Comment (yesmeck) [inline src/RangeCalendar.js]: I don't think change the return of `disabledTime` is a good idea. We should call this `getEndDisableTime` when CalendarPart renders.

Comment (kumaw) [reply src/RangeCalendar.js]: It isn't change the return of disabledTime. This is for when the hour isn‘t equal，disabledMinutes in getEndDisableTime will return [].But disabledMinutes should return use's disabledMinutes。

Comment (yesmeck) [reply src/RangeCalendar.js]: Hmm, I misunderstand the API.

File: src/RangeCalendar.js [hunk 2]
```diff
@@ -313,7 +313,7 @@ const RangeCalendar = createReactClass({
   getEndDisableTime() {
     const { selectedValue, value } = this.state;
     const { disabledTime } = this.props;
-    const userSettingDisabledTime = disabledTime(null, 'end') || {};
+    const userSettingDisabledTime = disabledTime(selectedValue, 'end') || {};
```
Comment (yesmeck) [inline src/RangeCalendar.js]: This is a breaking change.

Comment (yesmeck) [reply src/RangeCalendar.js]: Never mind.

**Comment by yesmeck** [comment]
Thanks.

File: src/RangeCalendar.js (hunk 2 above)
Comment (yesmeck) [reply src/RangeCalendar.js]: But it's not necessary to modify this line right?

**Comment by yesmeck** [comment]
Close https://github.com/ant-design/ant-design/issues/8915

**Pull request merged**